pkg/collector/labels.go

```diff
@@ -16,17 +16,29 @@ package collector
 
 import (
 	"fmt"
+	"regexp"
 
 	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 )
 
+func isFilteredLabel(label string, filterLabels []string) bool {
```

Comment:
This seems well-tested via the collector/daemonset test suites, but I think it does deserve a new table-test as part of the label's test suite.